plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        ntp_facts = facts['ansible_network_resources'].get('ntp')
+
+        if not ntp_facts:
+            return []
+        return ntp_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+        commands = list()
+        requests = list()
+
+        existing_ntp_facts = self.get_ntp_facts()
+
+        commands, requests = self.set_config(existing_ntp_facts)
+
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    if not re.search("code.*404", str(exc)):
+                        # Not "Resource not found"
+                        self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_ntp_facts = self.get_ntp_facts()
+
+        result['before'] = existing_ntp_facts
+        if result['changed']:
+            result['after'] = changed_ntp_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_ntp_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        if want is None:
+            want = []
+
+        have = existing_ntp_facts
+
+        resp = self.set_state(want, have)
+
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+
+        self.preprocess_want(want, state)
+
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :param want: the additive configuration as a dictionary
+        :param obj_in_have: the current configuration as a dictionary
+        :rtype: A list
```

Comment:
This function has no "rtype" parameter, so this line should be deleted.

Reply:
Done

Reply:
Verified.